DSharpPlus/Utilities.cs

```diff
@@ -322,6 +322,32 @@ internal static void LogTaskFault(this Task task, ILogger logger, LogLevel level
         value = kvp.Value;
     }
 
+    /// <summary>
+    /// Creates a snowflake from a given <see cref="DateTimeOffset"/>. This can be used to provide "timestamps" for methods
+    /// like <see cref="DiscordChannel.GetMessagesAfterAsync"/>.
+    /// </summary>
+    /// <param name="dateTimeOffset">DateTimeOffset to create a snowflake from.</param>
+    /// <returns>Returns a snowflake representing the given date and time.</returns>
+    public static ulong CreateSnowflake(DateTimeOffset dateTimeOffset)
+    {
+        var diff = dateTimeOffset - DiscordClient._discordEpoch;
```

Comment:
it might be worth calculating this diff from TotalMilliseconds - TimeSpan has some extra restrictions around how long a value can be

Reply:
like this?

```
var diff = dateTimeOffset.ToUnixTimeMilliseconds() - DiscordClient._discordEpoch.ToUnixTimeMilliseconds();
return (ulong)diff << 22;
```